File: api/models/tool.py
# models/tool.py
from pydantic import BaseModel
from typing import Optional

class Tool(BaseModel):
    id: Optional[str] = None
    model: str
    borrowDays: int
    status: str
    borrowedBy: Optional[str] = None
    description: Optional[str] = None

    class Config:
        extra = "ignore"

File: app/lib/models/tool.dart
class Tool {
  String? id;
  String model;
  int borrowDays;
  String status;
  String? borrowedBy; //null if status is available
  String? description;

  Tool({
    this.id,
    required this.model,
    required this.borrowDays,
    required this.status,
    this.borrowedBy,
    this.description,
  });

  factory Tool.fromJson(Map<String, dynamic> json) {
    return Tool(
      id: json['id'],
      model: json['model'],
      borrowDays: json['borrowDays'],
      status: json['status'],
      borrowedBy: json['borrowedBy'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'model': model,
      'borrowDays': borrowDays,
      'status': status,
      'borrowedBy': borrowedBy,
      'description': description,
    };
  }
}
